### ISSUE TITLE
Clarify that Java's UseContainerSupport doesn't currently work on CircleCI

### ISSUE THREAD

**Issue by danielcompton** [circleci/circleci-docs#3938]
# Locations
https://circleci.com/docs/2.0/java-oom/
https://circleci.com/blog/how-to-handle-java-oom-errors/

## Description

The blog post says:

> There’s help on the horizon, though: Java has a new(ish) ability to read the cgroup memory limits of your build’s Docker container, rather than (mis)reading the total memory of the entire machine. These new options should make it easier to get the JVM to use “most” of the memory on the machine, without going over.

New versions of Java (backported to JDK8) support reading memory limits from CGroups and setting their heap size based on a fraction of memory allocated to the container, rather than the total amount of ram in the system. - https://medium.com/adorsys/usecontainersupport-to-the-rescue-e77d6cfea712

**Docker**

You can see MaxHeapSize is set correctly when running the JDK under Docker:

```
docker run --rm -ti -m1000m circleci/clojure:openjdk-11-tools-deps java -XX:+PrintFlagsFinal -XX:MaxRAMPercentage=80.0 -version

...
   size_t MaxHeapSize                              = 838860800                                 {product} {ergonomic}
```

This sets `MaxHeapSize` to 800 Mebibytes.

**CircleCI**

However, if you run the same command (`java -XX:+PrintFlagsFinal -XX:MaxRAMPercentage=80.0 -version`) on CircleCI in a standard container, you get: 

```
   size_t MaxHeapSize                              = 32178700288                               {product} {ergonomic}
```

which is 32.18 gigabytes. This seems to be based on the memory on the machine, not the memory allocated to the container.

## Root cause (?)

This appears to be because of a bug/missing feature in Nomad: https://github.com/hashicorp/nomad/issues/5376.

It would be great if this was documented, it took me some time to track this down and talk with support to try and work out where the problem was.

Relates to #2988

**Comment by teesloane** [comment]
Hi there @danielcompton!

Sorry that you had to spend a while debugging this issue. I've created a PR to add a note about +useContainerSupport not being supported. I am not very familiar with Java and OOM issues; I've just added a line to note the lack of support for UseContainer, so if there are any other details we can add please don't hesitate to comment, open a new issue or PR.

**Comment by teesloane** [comment]
Closed by #3945

**Comment by danielcompton** [comment]
@teesloane it looks like this has been [fixed](https://github.com/hashicorp/nomad/issues/5376#issuecomment-790815724) in Nomad 0.9.0, so this note can probably be removed. I'm not using Circle CI at the moment so I can't test it to confirm, but you may want to take a look at it.
